### PR TITLE
PLAT-764 EmfImport: Improved Naming of Variables of CSV Files

### PR DESCRIPTION
**Pull request by thees-k**
It's possible to use variable names with more than one characters (see `EmfImportVariablesFinderTest` about the details).

Like in PLAT-723 (https://github.com/softicar/platform/pull/181), please use page "Servers" to try it out:

Create a .csv file with the following content and upload it:
```
$$Variable_A$$,127.0.0.1,$$Wooden_Mallet$$,username,secret-password,nix.com
test1,127.0.0.1,1234,$$gardenFence$$,secret-password,nix.com
$$variable_a$$,127.0.0.1,1234,username,secret-password,$$GardenFence$$
```

### CODE REVIEW

File: platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java [hunk 1]
```diff
@@ -420,6 +417,9 @@
 		.de("Wert '%s' liegt außerhalb des Wertebereichs");
 	I18n0 VALUE_CLASS = new I18n0("Value Class")//
 		.de("Klasse des Werts");
+	I18n0 VARIABLE_IDENTIFIERS_MAY_CONSIST_OF_ASCII_LETTERS_NUMBERS_FROM_0_TO_9_AND_THE_UNDERSCORE =
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java:420]:
```suggestion
	I18n0 VARIABLE_IDENTIFIERS_MAY_CONSIST_OF_ASCII_LETTERS_DIGITS_AND_THE_UNDERSCORE =
```

File: platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java [hunk 2]
```diff
@@ -420,6 +417,9 @@
...
+			new I18n0("Variable identifiers may consist of ASCII letters, numbers from 0 to 9 and the underscore.")//
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java:421]:
```suggestion
			new I18n0("Variable identifiers may consist of ASCII letters, digits and the underscore.")//
```

File: platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java [hunk 3]
```diff
@@ -420,6 +417,9 @@
...
+				.de("Variablenbezeichner dürfen aus ASCII-Buchstaben, Zahlen von 0 bis 9 und dem Unterstrich bestehen.");
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/EmfI18n.java:422]:
```suggestion
				.de("Variablenbezeichner dürfen aus ASCII-Buchstaben, Ziffern und dem Unterstrich bestehen.");
```

File: platform-emf/src/main/java/com/softicar/platform/emf/management/importing/upload/EmfImportUploadDiv.java [hunk 4]
```diff
@@ -35,8 +35,9 @@ public EmfImportUploadDiv(EmfImportPopup<R, P, S> popup) {
 				.addUnorderedListItem(EmfI18n.COLUMN_VALUES_MUST_BE_SEPARATED_BY_COMMAS)
 				.addUnorderedListItem(EmfI18n.INDIVIDUAL_COLUMN_VALUES_CAN_BE_ENCLOSED_BY_QUOTATION_MARKS_FOR_EXAMPLE_VALUE)
 				.addUnorderedListItem(
-					EmfI18n.YOU_CAN_DEFINE_VARIABLES_BY_ENCLOSING_THEM_WITH_DOUBLE_DOLLAR_SIGNS_FOR_EXAMPLE_ARG1_TO_DEFINE_VARIABLE_ARG2.toDisplay("$$a$$", "a"))
-				.addUnorderedListItem(EmfI18n.POSSIBLE_VARIABLE_IDENTIFIERS_ARE_ALL_ASCII_LOWERCASE_LETTERS_FROM_A_TO_Z)
+					EmfI18n.YOU_CAN_DEFINE_VARIABLES_BY_ENCLOSING_THEM_WITH_DOUBLE_DOLLAR_SIGNS_FOR_EXAMPLE_ARG1_TO_DEFINE_VARIABLE_ARG2
+						.toDisplay("$$a$$", "a"))
+				.addUnorderedListItem(EmfI18n.VARIABLE_IDENTIFIERS_MAY_CONSIST_OF_ASCII_LETTERS_NUMBERS_FROM_0_TO_9_AND_THE_UNDERSCORE)
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/management/importing/upload/EmfImportUploadDiv.java:40]:
```suggestion
				.addUnorderedListItem(EmfI18n.VARIABLE_IDENTIFIERS_MAY_CONSIST_OF_ASCII_LETTERS_DIGITS_AND_THE_UNDERSCORE)
```

File: platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java [hunk 5]
```diff
@@ -18,11 +21,13 @@ public EmfImportVariablesFinder(List<List<String>> rows) {
 
 	public SetMap<String, EmfImportVariableCoordinates> find() {
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java:22]: I think this method (and the class itself) deserves some Javadoc.
E.g., personally, I was surprised that we silently ignore malformed variable names.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java [hunk 6]
```diff
@@ -2,27 +2,46 @@
...
+	private final SetMap<String, EmfImportVariableCoordinates> variableDisplayNameToCoordinatesMap = new SetMap<>();
 
 	public EmfImportVariablesFinder(List<List<String>> rows) {
 
 		this.rows = rows;
 	}
 
+	/**
+	 * Finds variables and their coordinates in the given matrix of
+	 * {@link String} tokens.
```
Comment (alexander-schmidt) [inline platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java:32]: I think we need to add to this Javadoc block:
```
<p>
Invalid variable names will be silently ignored; they will not be added to the return value, and no exception will occur. Valid variable names match the following pattern: {@value #VARIABLE_PATTERN}
```

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java (hunk 6 above)
Comment (thees-k) [reply platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java:32]: added as suggested

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**